.github/workflows/ci.yml

```
@@ -0,0 +1,213 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  workflow_dispatch:
+    inputs:
+      save:
+        description: 'Save Filecoin parameters'
+        required: false
+        default: 'false'
+      publish:
+        description: 'Publish the static library'
+        required: false
+        default: 'false'
+      run-leak-detector:
+        description: 'Run the CGO leak detector'
+        required: false
+        default: 'false'
+      ref:
+        description: 'The ref to build'
+        required: false
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+permissions:
+  contents: read
+
+# Can we apply these to the entire workflow?
+env:
+  # Build the kernel only for the single architecture. This should reduce
+  # the overall compile-time significantly.
+  EC_GPU_CUDA_NVCC_ARGS: --fatbin --gpu-architecture=sm_75 --generate-code=arch=compute_75,code=sm_75
+  BELLMAN_CUDA_NVCC_ARGS: --fatbin --gpu-architecture=sm_75 --generate-code=arch=compute_75,code=sm_75
+  NEPTUNE_CUDA_NVCC_ARGS: --fatbin --gpu-architecture=sm_75 --generate-code=arch=compute_75,code=sm_75
+  DEBIAN_FRONTEND: noninteractive
+
+jobs:
+  check:
+    name: Check code style and run linters
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: ./.github/actions/configure-environment
+      - if: github.event.inputs.ref != ''
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          ref: ${{ github.event.inputs.ref }}
+      - name: Run shellcheck
+        run: shellcheck ./install-filcrypto
+      - name: Run cargo fmt
+        run: |
+          rustup component add rustfmt
+          cargo fmt --manifest-path ./rust/Cargo.toml --all -- --check
+      - name: Run cargo clippy
+        run: cd rust && cargo clippy --all-targets --features blst-portable,opencl -- -D warnings
+      - name: Run go fmt
+        # `! go fmt ./... 2>&1 | read"` doesn't work, this one does, thanks
+        # https://carsonip.me/posts/go-fmt-and-ci/
+        run: |
+          output=$(go fmt ./...)
+          echo "${output}"
+          test -z "${output}"
+      - name: Run various linters
+        run: |
+          go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.56.2
```

```suggestion
          go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.59.0
```

How about we jump up to the same version we've pinned lotus to. I don't think this introduces new failures, at least from my local testing.

---

Sure! Let's give it a go.